mvc/controllers/actions.rst

```diff
@@ -1,8 +1,63 @@
+
+What is a Controller
+--------------------
+
+In ASP.NET MVC, a `Controller` is used to define and group a set of actions. An `action` (or `action method`) is a method on a controller that handles incoming requests. In ASP.NET 5, a controller can be any instantiable class that ends in "Controller" or inherits from a class that ends with "Controller". By convention, controller classes may be placed within a root-level "Controllers" folder within the app, and may inherit from ``Microsoft.AspNet.Mvc.Controller``, but these are not required.
+
+Controllers support :doc:`dependency injection <dependency-injection>`, and should follow the `Explicit Dependencies Principle <http://deviq.com/explicit-dependencies-principle>`_ and request any dependencies their actions require through their constructor. Controllers provide a logical means of grouping similar actions together, allowing common sets of rules (e.g. routing, caching, authorization) to be applied collectively.
+
+Within the Model-View-Controller pattern, a Controller is responsible for the initial processing of the request and instantiation of the Model. Business decisions should ideally be performed within the Model. The Controller then takes the result of the Model's processing (if any), chooses which View should be returned, and returns that View along with any data that View requires. Learn more: :doc:`../overview`.
+
+.. tip:: The Controller is a `UI level` abstraction. Its responsibility is to ensure incoming request data is valid and to choose which View (or result) should be returned. In well-factored apps it will not include data access or business logic, but instead will delegate to services handling these responsibilities.
+ 
+Defining Actions
+----------------
+Any public method on a controller type is an action. Parameters on actions are bound to request data and validated using :doc:`model binding <../models/model-binding>`. Any action that accepts parameters should verify that the incoming data is valid by checking ``ModelState.IsValid``.
+
+.. warning:: Action methods that accept parameters should always check ``ModelState.IsValid`` as part of their processing.
+
+Action methods should contain logic for mapping an incoming request to a business concern. These business concerns should typically be represented as services that your controller accesses through dependency injection. Actions then map the result of the business action to an application state.
+
+Actions can return an ``IActionResult`` that produces a response. You can either create and return your own ``ActionResult`` directly, or you can use one of the many helper methods available on the Controller base class.
+
+Controller Helper Methods
+#########################
+
+Inheriting from the base ``Controller`` class provides the following helper methods for returning various kinds of responses.
+
+View
+	Returns a View that uses a model to render HTML. Example: ``return View(customer);``
```

Comment:
On IActionResult or ActionResult. ( I asked doug Bunting) - he responded:
Those aren’t return types, they’re `ActionResult` subclasses that can be instantiated and returned e.g. using `View(…)` or `Json(…)` or … And, yes, MVC 6 documentation should cover all but `JavaScriptResult`, which we haven’t brought forward 

Note actions can also return instances of model types or void. Might be an advanced topic however. Daniel, where does `return new MyModel(…);` fit?

	about why the template generated return type has changed?
We added `IActionResult` because we thought it was worthwhile. Turns out `ActionResult` remains the only implementation of that interface.